examples/api/lib/widgets/safe_area/safe_area.0.dart

```diff
@@ -0,0 +1,288 @@
+      thumbShape: const RoundSliderThumbShape(enabledThumbRadius: 12),
+    ),
+    scaffoldBackgroundColor: const Color(0xFFD0FFE8),
+    listTileTheme: const ListTileThemeData(
+      tileColor: Colors.white70,
+      contentPadding: EdgeInsets.all(6.0),
+    ),
+    appBarTheme: AppBarTheme(
+      backgroundColor: colors.secondary,
+      foregroundColor: colors.onSecondary,
+    ),
+  );
+
+  static final AppBar appBar = AppBar(title: const Text('SafeArea Demo'));
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: theme,
+      debugShowCheckedModeBanner: false,
+      home: Builder(
+        builder: (BuildContext context) => Scaffold(
+          appBar: Toggle.appBar.of(context) ? appBar : null,
+          body: const DefaultTextStyle(
+            style: TextStyle(
+              fontSize: 16,
+              fontWeight: FontWeight.w600,
+              color: Colors.black,
+            ),
+            child: Center(
+              child: SafeAreaExample(),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class SafeAreaExample extends StatelessWidget {
+  const SafeAreaExample({super.key});
+
+  static final Widget controls = Column(
+    children: <Widget>[
+      const SizedBox(height: 14),
+      Builder(
+        builder: (BuildContext context) => Text(
+          Toggle.safeArea.of(context) ? 'safe area!' : 'no safe area',
+          style: const TextStyle(fontSize: 24),
+        ),
+      ),
+      const Spacer(flex: 2),
+      for (final Value data in Value.allValues) ...data.controls,
+    ],
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    final bool hasSafeArea = Toggle.safeArea.of(context);
+
+    return SafeArea(
+      top: hasSafeArea,
+      bottom: hasSafeArea,
+      left: hasSafeArea,
+      right: hasSafeArea,
+      child: controls,
+    );
+  }
+}
+
+sealed class Value implements Enum {
+  Object _getValue(covariant Model<Value> model);
+
+  List<Widget> get controls;
+
+  static const List<Value> allValues = <Value>[...Inset.values, ...Toggle.values];
+}
+
+enum Inset implements Value {
+  top('top notch'),
+  sides('side padding'),
+  bottom('bottom indicator');
+
+  const Inset(this.label);
+
+  final String label;
+
+  @override
+  double _getValue(_InsetModel model) => switch (this) {
+    top    => model.insets.top,
+    sides  => model.insets.left,
+    bottom => model.insets.bottom,
+  };
+
+  double of(BuildContext context) => _getValue(Model.of<_InsetModel>(context, this));
+
+  @override
+  List<Widget> get controls => <Widget>[
+    Text(label),
+    Builder(
+      builder: (BuildContext context) => Slider(
+        max: 50,
+        value: of(context),
+        onChanged: (double newValue) {
+          InsetsState.instance.newInsets(this, newValue);
+        },
+      ),
+    ),
+    const Spacer(),
+  ];
+}
+
+enum Toggle implements Value {
+  appBar('show AppBar?'),
+  safeArea('wrap body with SafeArea?');
```

Comment:
What do you think of capitalizing this text?

Also, what do you think of clarifying that these update a `Scaffold`?

Something like:

```suggestion
  appBar('Add Scaffold appBar?'),
  safeArea('Wrap Scaffold body with SafeArea?');
```

Reply:
The capitalization was a stylistic choice… which means we might as well follow the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-correct-grammar)!

Part of me feels that it's obvious that these apply to the Scaffold (a Scaffold with an app bar & body is one of the first things a developer sees after running `flutter create`), but there's not much of a downside to including it here 👍